modules/caddytls/connpolicy.go

```diff
@@ -232,6 +232,9 @@ func (p *ConnectionPolicy) buildStandardTLSConfig(ctx caddy.Context) error {
 				cfg.CertSelection = p.CertSelection
 			}
 			cfg.DefaultServerName = p.DefaultSNI
+			if hello.ServerName == "" && p.DefaultSNI != "" {
```

Comment:
The second validation is kinda redundant right? Worst case scenario you're just setting "" = ""